pkg/operator/controller/dns_status_test.go

```diff
@@ -212,140 +223,184 @@ func TestComputeDNSDegradedCondition(t *testing.T) {
+			expectRequeue: false,
 		},
 		{
-			name:         "too few pods DNS pods available (percentage)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(100, 89, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(100, 100, intstr.FromString("33%")),
-			expected:     operatorv1.ConditionTrue,
-		},
-		{
-			name:         "node-resolver pods unavailable is ok (percentage)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(100, 100, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(100, 65, intstr.FromString("33%")),
-			expected:     operatorv1.ConditionFalse,
-		},
-		{
-			name:         "too few DNS pods available (integer)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 4, intstr.FromInt(1)),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.FromInt(1)),
-			expected:     operatorv1.ConditionTrue,
+			name:          "too few DNS pods available (percentage)",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(100, 89, intstr.FromString("10%")),
+			oldCondition:  defaultOldCondition,
+			expected:      operatorv1.ConditionFalse,
+			expectAfter:   time.Minute * 5,
+			expectRequeue: true,
 		},
 		{
-			name:         "node-resolver pods unavailable is ok (integer)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 6, intstr.FromInt(1)),
-			nrDaemonset:  makeDaemonSet(6, 4, intstr.FromInt(1)),
-			expected:     operatorv1.ConditionFalse,
+			name:          "too few DNS pods available (integer)",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(6, 4, intstr.FromInt(1)),
+			oldCondition:  defaultOldCondition,
+			expected:      operatorv1.ConditionFalse,
+			expectAfter:   time.Minute * 5,
+			expectRequeue: true,
 		},
 		{
 			name:         "enough available (percentage)",
 			clusterIP:    "172.30.0.10",
 			dnsDaemonset: makeDaemonSet(100, 90, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(100, 67, intstr.FromString("33%")),
+			oldCondition: defaultOldCondition,
 			expected:     operatorv1.ConditionFalse,
 		},
 		{
 			name:         "enough available (integer)",
 			clusterIP:    "172.30.0.10",
 			dnsDaemonset: makeDaemonSet(6, 5, intstr.FromInt(1)),
-			nrDaemonset:  makeDaemonSet(6, 5, intstr.FromInt(1)),
+			oldCondition: defaultOldCondition,
 			expected:     operatorv1.ConditionFalse,
 		},
 		{
 			name:         "all available",
 			clusterIP:    "172.30.0.10",
 			dnsDaemonset: makeDaemonSet(6, 6, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.FromString("33%")),
+			oldCondition: defaultOldCondition,
 			expected:     operatorv1.ConditionFalse,
 		},
+		{
+			name:          "not enough DNS pods available, check requeue",
+			clusterIP:     "172.30.0.10",
+			dnsDaemonset:  makeDaemonSet(10, 8, intstr.FromString("10%")),
+			oldCondition:  defaultOldCondition,
+			expected:      operatorv1.ConditionFalse,
+			expectAfter:   time.Minute * 5,
+			expectRequeue: true,
+		},
```

Comment:
This is covered by the "too few DNS pods available (percentage)" test case.

Reply:
In this case I"m explicitly checking for a requeue value.

Reply:
What do you mean?  Both "too few DNS pods available (percentage)" and "not enough DNS pods available, check requeue" have the following:
```go
			expectAfter:   time.Minute * 5,
			expectRequeue: true,
```
The two test cases only differ in the name and daemonset, namely one has the following:
```go
			dnsDaemonset:  makeDaemonSet(100, 89, intstr.FromString("10%")),
```
and the other has the following:
```go
			dnsDaemonset:  makeDaemonSet(10, 8, intstr.FromString("10%")),
```
which ultimately should both lead to the same degraded-but-with-grace outcome.

Reply:
Removed